src/theory/quantifiers/sygus/sygus_interpol.cpp

```diff
@@ -204,18 +204,122 @@ TypeNode SygusInterpol::setSynthGrammar(const TypeNode& itpGType,
+  Trace("sygus-interpol-debug")
+      << "Make interpolation predicate app..." << std::endl;
+  std::vector<Node> ichildren;
+  ichildren.push_back(itp);
+  ichildren.insert(ichildren.end(), d_varsShared.begin(), d_varsShared.end());
+  Node itpApp =
+      d_varsShared.empty() ? itp : nm->mkNode(kind::APPLY_UF, ichildren);
+  Trace("sygus-interpol-debug") << "...finish" << std::endl;
+
+  // set the sygus bound variable list
+  Trace("sygus-interpol-debug") << "Set attributes..." << std::endl;
+  itp.setAttribute(theory::SygusSynthFunVarListAttribute(), d_ibvlShared);
+  // sygus attribute
+  Node sygusVar = nm->mkSkolem("sygus", nm->booleanType());
+  theory::SygusAttribute ca;
+  sygusVar.setAttribute(ca, true);
+  Node instAttr = nm->mkNode(kind::INST_ATTRIBUTE, sygusVar);
+  std::vector<Node> iplc;
+  iplc.push_back(instAttr);
+  Node instAttrList = nm->mkNode(kind::INST_PATTERN_LIST, iplc);
+  Trace("sygus-interpol-debug") << "...finish" << std::endl;
+
+  // Fa( x )
+  Trace("sygus-interpol-debug") << "Make conjecture body..." << std::endl;
+  Node Fa = axioms.size() == 1 ? axioms[0] : nm->mkNode(kind::AND, axioms);
+  Trace("sygus-interpol-debug") << "Fa before substitution: " << Fa << std::endl
+                                << std::endl;
+  Fa =
+      Fa.substitute(d_syms.begin(), d_syms.end(), d_vars.begin(), d_vars.end());
+  Trace("sygus-interpol-debug") << "Fa after substitution: " << Fa << std::endl
+                                << std::endl;
+  Trace("sygus-interpol-debug") << "itpApp: " << itpApp << std::endl
+                                << std::endl;
+  // Fa( x ) => A( x )
+  Node firstImplication = nm->mkNode(kind::IMPLIES, Fa, itpApp);
+  Trace("sygus-interpol-debug")
+      << "first implication: " << firstImplication << std::endl
+      << std::endl;
+  // A( x ) => Fc( x )
+  Node Fc = conj;
+  Fc =
+      Fc.substitute(d_syms.begin(), d_syms.end(), d_vars.begin(), d_vars.end());
+  Node secondImplication = nm->mkNode(kind::IMPLIES, itpApp, Fc);
+  Trace("sygus-interpol-debug")
+      << "second implication: " << secondImplication << std::endl
+      << std::endl;
+  // Fa( x ) => A( x ) ^ A( x ) => Fc( x )
+  Node constraint = nm->mkNode(kind::AND, firstImplication, secondImplication);
+  Trace("sygus-interpol-debug") << constraint << "...finish" << std::endl;
+  constraint = theory::Rewriter::rewrite(constraint);
+
+  // forall A. exists x. ~( Fa( x ) => A( x ) ^ A( x ) => Fc( x ) )
+  Trace("sygus-interpol-debug") << "Make conjecture..." << std::endl;
+  d_sygusConj = constraint;
+  Trace("sygus-interpol-debug") << "...finish" << std::endl;
+  Trace("sygus-interpol") << "Generate: " << d_sygusConj << std::endl;
 }
 
 bool SygusInterpol::findInterpol(Expr& interpol, Node itp)
 {
+  // get the synthesis solution
+  std::map<Expr, Expr> sols;
+  d_subSolver->getSynthSolutions(sols);
+  Assert(sols.size() == 1);
+  std::map<Expr, Expr>::iterator its = sols.find(itp.toExpr());
+  if (its != sols.end())
```

Comment:
You could invert this test to throw the exception here and return. Avoids the extra indentation for the large block of code.